### PR TITLE
Update tech dropdown 2397

### LINKED ISSUE #2397: Update the Technologies dropdown in the Projects page to match the Figma design
### Overview
We need to update the ~` Technology`~ `Languages / Technologies` dropdown menu on the Projects page to match the newly created Figma design so that we can provide a better user experience for our end-user.

### Details
With the addition of Languages into the now Languages / Technologies dropdown (formerly Technologies dropdown), this issue had to be modified to take into account the inclusion of languages in the dropdown. @IsaacDesigns from Design Team said to go ahead with implementing Version C of the desktop design and implementing the mobile design. (See https://github.com/hackforla/website/pull/2672#issuecomment-1013904367 for more info.)

<details>
  <summary><strong>Click here</strong> to see the current projects page `Languages / Technologies` dropdown on the live website for desktop and mobile(<strong>before changes</strong>)</summary>

**On desktop**  
desktop-project-page-tech-lang-dropdown

**On mobile**
mobile-project-page-tech-lang-dropdown

</details>

<details>
  <summary><strong>Click here</strong> to see the proposed projects page `Languages / Technologies` dropdown Figma designs for desktop and mobile (<strong>after changes</strong>)</summary>

**Desktop design Version C**
version-c-tech-lang-dropdown

**Mobile design**
mobile-tech-lang-dropdown

</details>

<details>
  <summary>Outdated screenshots (Do not use these)</summary>
  <details>
    <summary><strong>Click here</strong> to see the current projects page `Technologies` dropdown (<strong>before changes</strong>)</summary>
  
  projects-dropdown-before
  
  </details>
  
  <details>
    <summary><strong>Click here</strong> to see the proposed projects page `Technologies` dropdown Figma design (<strong>after changes</strong>)</summary>
  
  projects-dropdown-after
  </details>

</details>

### Action Items
- [x] Refactor the CSS for the desktop version of the ~` Technologies`~ `Languages / Technologies` dropdown to be 768px wide and the checkboxes create the appropriate columns to fit all technologies and languages. Also, include the scrollbar.
- [x] Ensure the dropdown matches [~Version A~ Version C of the Projects page Figma design](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=15313%3A101641)
- [x] Refactor the CSS for the mobile version of the `Languages / Technologies` dropdown to include a scrollbar
- [x] Ensure the dropdown matches the [Projects page Figma design for mobile](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=15902%3A100503)
- [x] When complete, submit a pull request with the proposed changes
- [x] Make sure the page looks fine for smaller viewport screens as well, such as tablet, mobile (responsiveness)

### Resources/Instructions
- ~Version A~ Version C and mobile design within https://github.com/hackforla/website/issues/2069#issuecomment-927388942
~ [Figma- Projects page dropdown (~Version A~ Version C and mobile)](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=532%3A1454)

### PR DESCRIPTION
**Pull request by anthonysim**
Fixes #2397 

### What changes did you make and why did you make them ?

  - Added media query for min-width 768px. I added css grid at a width of 748px so the new dropdown would fit properly on an IPAD and IPAD Pro. The original issue did not account for other sizes except a desktop so I had to make sure everything "looked" good on an IPAD and IPAD Pro.
  - Added media query for min-width 1024px. I added a width of 768px as the original issue stated.
  - Ensure the dropdown matches [Version A of the Projects page Figma design](https://github.com/hackforla/website/issues/2069#issuecomment-927388942). The issue this PR is associated with did not state to rearrange the dropdowns as well as shrink them. TBD. 
  - Version C Design was implemented along with scrollbars for mobile and full screen. 
  
- See additional notes/comments by merger @JessicaLucindaCheng: https://github.com/hackforla/website/pull/2672#issuecomment-1027453766  

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
</details>

<details>
<summary>Visuals after Version C was implemented</summary>
  
  
</details>

### CODE REVIEW

**Comment by JessicaLucindaCheng** [comment]
- Availability: 6 hours
- Review ETA: Wed, Jan 12, 2022

**Comment by lopezpedres** [comment]
Availability: 10 hours
Review ETA: Wed, Jan 11, 2022

**Comment by lopezpedres** [comment]
Hi @anthonysim ! REALLY GOOD JOB. The dropdown looks really well. It ran really nice on the desktop, iPad, and iPad Pro view.
There are two main points that I just want to highlight (I am not entirely sure If they are part of this issue, we might need to create new ones)

1. According to the version A of the Figma design, the dropdown arrow should be pointing up when the dropdown is selected

    <details>
    <summary>Figma Version A</summary>
   
    Dropwdown Arrow

    </details>
    <details>
    <summary>Antony's version</summary>
  
    PR version

    </details>

2. The mobile view should have a scrolling bar once the dropdown menu is active
     <details>
    <summary>Figma Mobile Version</summary>
       
    image

    </details>
    <details>
    <summary>Antony's version</summary>
  
    mobile dropdown

    </details>

@JessicaLucindaCheng please let me know what you think about these points. Otherwise, ILGTM and ready to be merged :)

**Comment by lopezpedres** [comment]
> Hi @anthonysim ! REALLY GOOD JOB. The dropdown looks really well. It ran really nice on the desktop, iPad, and iPad Pro view. There are two main points that I just want to highlight (I am not entirely sure If they are part of this issue, we might need to create new ones)
> 
> 1. According to the version A of the Figma design, the dropdown arrow should be pointing up when the dropdown is selected
>     
>     Figma Version A
>    Dropwdown Arrow
>     
>     
>     Antony's version
>    PR version
> 2. The mobile view should have a scrolling bar once the dropdown menu is active
>      
>     Figma Mobile Version
>    image
>     
>     
>     Antony's version
>    mobile dropdown
> 
> @JessicaLucindaCheng please let me know what you think about these points. Otherwise, ILGTM and ready to be merged :)

Sorry guys! I just saw Anthony's comment about the mobile version. Please ignore the second point :) @anthonysim @JessicaLucindaCheng

**Comment by JessicaLucindaCheng** [comment]
@lopezpedres In the current website, all the dropdowns (Status, Technologies/Languages, Program Areas) have the arrows pointing down. I think the arrow of the dropdown is fine as is in terms of this issue. However, we can bring it up with the design team and ask them. We may want to ask design for an updated projects page, so all of us are on the same page in terms of what is wanted currently, including with the addition of languages from pr https://github.com/hackforla/website/pull/2646.

**Comment by lopezpedres** [comment]
@JessicaLucindaCheng Awesome! Thanks for replying :)

**Review by lopezpedres — DISMISSED** [review]
Nicely done @anthonysim !

File: _sass/components/_project-filter.scss [hunk 1]
```diff
@@ -7,10 +7,9 @@ a.filter-item {
 //filter container
 ul.filter-list {
   background: $color-pink;
-  display: grid;
-  list-style: none;
   padding-left: 0;
-  grid-auto-flow: column;
+  list-style: none;
+  display: grid;
```
Comment (JessicaLucindaCheng) [inline _sass/components/_project-filter.scss:12]: This was originally on lines 10 and 11. Just wondering why did you decide to move them?

File: _sass/components/_project-filter.scss [hunk 2]
```diff
@@ -168,6 +167,25 @@ a.clear-filter-tags {
   }
 }
 
+// sizing for technologies dropdown up to an ipad
+@media (min-width: 768px) {
```
Comment (JessicaLucindaCheng) [inline _sass/components/_project-filter.scss]: From https://github.com/hackforla/website/blob/gh-pages/_sass/variables/_layout.scss, please use one of the variables there for creating your media queries. For example, this line of code could become `@media ($bp-below-tablet) {`.

File: _sass/components/_project-filter.scss [hunk 3]
```diff
@@ -168,6 +167,25 @@ a.clear-filter-tags {
...
+    grid-auto-flow: column; // flows the data from top to bottom rather than left to right
+    grid-template-rows: repeat(10, 1fr); // ensures the list is 10 rows
+    list-style: none;
+    width: 748px;
+    left: -18em;
+  }
+}
+
+// sizing for technologies dropdown from an ipad and up
+@media (min-width: 1024px) {
```
Comment (JessicaLucindaCheng) [inline _sass/components/_project-filter.scss]: Would it be possible to use one of the variables in https://github.com/hackforla/website/blob/gh-pages/_sass/variables/_layout.scss for writing this media query?

**Review by JessicaLucindaCheng — CHANGES_REQUESTED** [review]
Thank you for working on this issue! Your changes along with the recent inclusion of languages into the technologies dropdown look like this:

wider-dropdown-anthony-sim-update-tech-dropdown-2397

Please update your forked repo and then merge those updates into your branch. Then, please make changes so that the menu includes the languages and is still 4 columns. Thank you.

**Comment by JessicaLucindaCheng** [comment]
@lopezpedres I see in your screenshots that you see 4 columns for the technologies dropdown. When you were viewing it, did it include languages?

File: _sass/components/_project-filter.scss (hunk 2 above)
Comment (anthonysim) [reply _sass/components/_project-filter.scss]: @JessicaLucindaCheng , sure will do. Ended up using `$bp-tablet-up` instead for min-width.

File: _sass/components/_project-filter.scss (hunk 3 above)
Comment (anthonysim) [reply _sass/components/_project-filter.scss]: @JessicaLucindaCheng, yeah, I can try and use one of the variables. I'll give it a shot and push up the change.

File: _sass/components/_project-filter.scss (hunk 1 above)
Comment (anthonysim) [reply _sass/components/_project-filter.scss:12]: @JessicaLucindaCheng , I just thought keeping all the stuff related to css grid close together made more sense. To see the grid, column gap, then grid template columns. It doesn't change anything, but made it easier to read.

**Comment by anthonysim** [comment]
@JessicaLucindaCheng @lopezpedres, I just changed the media queries to use variables.

File: _sass/components/_project-filter.scss (hunk 2 above)
Comment (JessicaLucindaCheng) [reply _sass/components/_project-filter.scss]: My bad. You are correct it should be $bp-tablet-up

**Comment by anthonysim** [comment]
> Thank you for working on this issue! Your changes along with the recent inclusion of languages into the technologies dropdown look like this:
> 
> wider-dropdown-anthony-sim-update-tech-dropdown-2397
> 
> Please update your forked repo and then merge those updates into your branch. Then, please make changes so that the menu includes the languages and is still 4 columns. Thank you.

So with the inclusion of the new technologies, I don't think I can follow Version A 
[Figma- Projects page dropdown (Version A)](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=15313%3A103303) as the original issue (#2397) stated. I'm assuming I should do Version C now and just do the scrollbar for mobile and desktop?

**Comment by JessicaLucindaCheng** [comment]
@anthonysim I realized I gave you the wrong formatting for using the variables in the media queries. Sorry for the confusion.
 
As for which Figma version to follow, let's hold off on making changes until we can get some clarification from Saumil and Tomas on what should be done next.

**Comment by JessicaLucindaCheng** [comment]
@SAUMILDHANKAR @R-Tomas-Gonzalez
See comment above: https://github.com/hackforla/website/pull/2672#issuecomment-1009649128

**Comment by lopezpedres** [comment]
> @lopezpedres I see in your screenshots that you see 4 columns for the technologies dropdown. When you were viewing it, did it include languages?

@JessicaLucindaCheng It did not. I'll attach the full screenshot:

image

**Comment by JessicaLucindaCheng** [comment]
> > @lopezpedres I see in your screenshots that you see 4 columns for the technologies dropdown. When you were viewing it, did it include languages?
> 
> @JessicaLucindaCheng It did not. I'll attach the full screenshot:
> 
> image

@lopezpedres Okay. The languages were recently merged into the Hack for LA repo and I updated my forked gh-pages before using the git commands (see comment: https://github.com/hackforla/website/pull/2672#issuecomment-1007876122) to do it. I think that's why we are seeing different things.

**Comment by JessicaLucindaCheng** [comment]
> So with the inclusion of the new technologies, I don't think I can follow Version A [Figma- Projects page dropdown (Version A)](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=15313%3A103303) as the original issue (#2397) stated. I'm assuming I should do Version C now and just do the scrollbar for mobile and desktop?

@anthonysim
@SAUMILDHANKAR spoke with @IsaacDesigns and Isaac said to go ahead with implementing Version C with the scrollbar for mobile and desktop in the Figma design. @JessicaLucindaCheng will update the issue to reflect this change.

**Comment by anthonysim** [comment]
@JessicaLucindaCheng @lopezpedres, I added the scrollbars for the mobile and desktop. 

Screen Shot 2022-01-16 at 10 46 22 AM
Screen Shot 2022-01-16 at 10 47 19 AM

**Review by lopezpedres — DISMISSED** [review]
@anthonysim Nicely done! LGTM Thanks for working in this issue Anthony!

**Comment by JessicaLucindaCheng** [comment]
- Availability: 6 hours
- Re-review ETA: Thu, Jan 20, 2022

**Review by JessicaLucindaCheng — CHANGES_REQUESTED** [review]
@anthonysim Thanks for all your work on this issue! In developer tools on my browser, the Surface Duo size screen (540px wide) shows the desktop version of the `Languages / Technologies` dropdown. It should show the mobile version. Please fix this.

This is what I see:
tech-dropdown-surface-duo-540px-screen

**Comment by anthonysim** [comment]
> > So with the inclusion of the new technologies, I don't think I can follow Version A [Figma- Projects page dropdown (Version A)](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=15313%3A103303) as the original issue (#2397) stated. I'm assuming I should do Version C now and just do the scrollbar for mobile and desktop?
> 
> @anthonysim @SAUMILDHANKAR spoke with @IsaacDesigns and Isaac said to go ahead with implementing Version C with the scrollbar for mobile and desktop in the Figma design. @JessicaLucindaCheng will update the issue to reflect this change.

@lopezpedres @JessicaLucindaCheng, I made changes to account for the Surface Duo 540px screen size. 
Screen Shot 2022-01-21 at 7 04 29 PM

**Comment by JessicaLucindaCheng** [comment]
- Availability: 6 hours
- Re-review ETA: Tuesday, January 25, 2022

File: _sass/components/_project-filter.scss [hunk 4]
```diff
@@ -165,6 +164,36 @@ a.clear-filter-tags {
...
+    list-style: none;
+    left: -17em;
+    width: 718px;
+    height: 380px;
+    overflow: auto;
+  }
+}
+
+// iPad Air to Desktop
+@media #{$bp-ipadair-up} {
```
Comment (JessicaLucindaCheng) [inline _sass/components/_project-filter.scss]: Could you use $bp-desktop-up instead?

**Review by JessicaLucindaCheng — CHANGES_REQUESTED** [review]
@anthonysim Besides my inline comment request for change, good job implementing Version C and mobile designs!

File: _sass/components/_project-filter.scss (hunk 4 above)
Comment (anthonysim) [reply _sass/components/_project-filter.scss]: @lopezpedres @JessicaLucindaCheng, sounds good, changed `$bp-ipadair-up` to `$bp-desktop-up`!

**Comment by JessicaLucindaCheng** [comment]
- Availability; 6 hours
- Re-review ETA: Thursday, January 27, 2022

**Comment by JessicaLucindaCheng** [comment]
@anthonysim Sorry, I don't have time to review it today. So, I will be delaying my review by one day.
- Availability: 6 hours
- ETA: Friday, January 28, 2022

**Review by JessicaLucindaCheng — CHANGES_REQUESTED** [review]
@anthonysim The work you did to make the menus work for desktop, tablet, and mobile look fantastic. Thank you for all your hardwork!

However, I have a few more requested changes:
1) We use PR descriptions/posts to look back at the codebase commit history and see when and why changes were made. With that in mind, could you clarify the actions you took in https://github.com/hackforla/website/pull/2672#issue-1096819720? Also, can you just have one final picture instead of two after pictures? Also, I noticed you used `em` for the SCSS property `left` instead of `px` you may want to include info for your reasoning for that. Also, you may want to include your reasoning for choosing a shorter height than 493px that was recommended for the desktop dropdown in the Figma design. (I think they were good choices but I'm just wondering your reasons for doing so.) 

2) For the _sass/variables/_layout.scss, I like how you are thinking for making this easier to understand but it may be confusing to include those changes with this issue because the issue has nothing to with the layouts.scss page. If you are interested, you could write up a small issue to include comments to make that page easier to read for developers and we can discuss this further if you are interested in doing so.  For now, could you remove the comments you added to the layout.scss file so it was the way it was before?

If you have questions about or want to discuss any of the above, please let me know and we can find a time to meet. Again, thank you for all your hard work.

**Comment by anthonysim** [comment]
@SAUMILDHANKAR @lopezpedres @JessicaLucindaCheng, hi everyone, there is a lot of back and forth on this PR, my solution seems to not be solving the issue so I'm going to close out the PR, place the issue associated with this PR back to the backlog under projects, and unassign myself from the issue. Maybe someone else can take this and give it a try instead.

**Comment by JessicaLucindaCheng** [comment]
@anthonysim I apologize if my request for changes suggested your solution did not work because your solution does work but needed some small changes. I understand that you are frustrated since you have been working on this issue for so long. However, I think we can use all the hard work you put into your pull request so far and I would like to merge what you have so far by reopening your pr to do so. You won't have to continue work on this issue as I will assign myself to take over the issue you were working on and just continue where you left off.

**Review by JessicaLucindaCheng — APPROVED** [review]
Merging pr as is and will be fixing it in a different branch.

**Comment by JessicaLucindaCheng** [comment]
## Overview
These notes are from me (@JessicaLucindaCheng), the person who merged the pr. In addition, I am taking up Anthony's issue because he has left Hack for LA and was almost done with the issue. Below are my notes about the implementation and changes made by Anthony.

### Final images of Desktop version C and the mobile version implemented by Anthony Sim

<details>
  <summary>Implemented Version C</summary>
  
  desktop-project-tech-lang-filter

</details>

<details>
  <summary>Implemented mobile version with scrollbar (Note: Scrollbar not visible in the screenshot but is there)</summary>

  mobile-project-tech-lang-filter

</details>


### Notes
- Differences I noticed from the Figma design
  - For the desktop version, I noticed em for the SCSS property left instead of using px for centering the tech dropdown.
  - Also, for the desktop version, the height of the dropdown is 380px for tablet and desktop views, which is a shorter height than 493px that was recommended for the desktop dropdown in the Figma design. I think this was done so that the dropdown fits in a desktop or tablet view while taking into consideration the space the fixed header navigation takes up on a screen.
  - In addition, the width of the dropdown was set to 718px instead of 768px for a tablet size screen. I think this was done because the width of a common tablet is set to 768px in [_sass/variables/_layout](https://github.com/hackforla/website/blob/7755e2cdec34deef4962736639177995faeeeb7c/_sass/variables/_layout.scss) and using the variable `$bp-tablet-up` in a media query for it would require the width of the dropdown to be smaller than 768px.

- Moreover, the _sass/variables/_layout.scss has added comments that are unrelated to the associated issue.

**Pull request merged**